### PR TITLE
add wla-dx and rgbds

### PR DESCRIPTION
**Pull request by matthewbauer**
###### Things done:
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- [x] Built on platform(s): OSX
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

Fixes issue #<insert id>

cc @<maintainer>

---

_Please note, that points are not mandatory, but rather desired._

### CODE REVIEW

**Comment by joachifm** [comment]
Hi. Would you please rename the commits to `pkgname: init at version`?

File: pkgs/development/compilers/wla-dx/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{stdenv, fetchFromGitHub, cmake}:
+
+stdenv.mkDerivation rec {
+  name = "wla-dx-${rev}";
+  rev = "8189fe8d5620584ea16563875ff3c5430527c86a";
```
Comment (joachifm) [inline pkgs/development/compilers/wla-dx/default.nix]: Is there a reason to use HEAD rather than v9.6? (apart from it being a bit old, that is)

Comment (matthewbauer) [inline pkgs/development/compilers/wla-dx/default.nix]: The 9.6 version uses a different build system. I could never get it to work.

File: pkgs/development/compilers/rgbds/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+stdenv.mkDerivation rec {
+  name = "rgbds-${version}";
+  version = "0.2.4";
+  src = fetchFromGitHub {
+    owner = "bentley";
+    repo = "rgbds";
+    rev = "v${version}";
+    sha256 = "0dwq0p9g1lci8sm12a2rfk0g33z2vr75x78zdf1g84djwbz8ipc6";
+  };
+  buildInputs = [ yacc ];
```
Comment (joachifm) [inline pkgs/development/compilers/rgbds/default.nix]: Is this build-time only? If so perhaps `nativeBuildInputs` makes sense. What do you think?

File: pkgs/development/compilers/rgbds/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  installFlags = "PREFIX=\${out}";
+
+  meta = with stdenv.lib; {
+    homepage = "https://www.anjbe.name/rgbds/";
+    description = "an assembler/linker package that produces Game Boy programs";
```
Comment (joachifm) [inline pkgs/development/compilers/rgbds/default.nix]: Please capitalize the first word of the description.

File: pkgs/development/compilers/rgbds/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    owner = "bentley";
+    repo = "rgbds";
+    rev = "v${version}";
+    sha256 = "0dwq0p9g1lci8sm12a2rfk0g33z2vr75x78zdf1g84djwbz8ipc6";
+  };
+  buildInputs = [ yacc ];
+  installFlags = "PREFIX=\${out}";
+
+  meta = with stdenv.lib; {
+    homepage = "https://www.anjbe.name/rgbds/";
```
Comment (joachifm) [inline pkgs/development/compilers/rgbds/default.nix]: These quotes are strictly unnecessary

**Comment by joachifm** [comment]
That's all the nitpicks I could come up with right now. LGTM.

File: pkgs/development/compilers/rgbds/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+stdenv.mkDerivation rec {
+  name = "rgbds-${version}";
+  version = "0.2.4";
+  src = fetchFromGitHub {
+    owner = "bentley";
+    repo = "rgbds";
+    rev = "v${version}";
+    sha256 = "0dwq0p9g1lci8sm12a2rfk0g33z2vr75x78zdf1g84djwbz8ipc6";
+  };
+  buildInputs = [ ];
```
Comment (joachifm) [inline pkgs/development/compilers/rgbds/default.nix]: Ach, one more thing, is this strictly necessary?

Comment (matthewbauer) [inline pkgs/development/compilers/rgbds/default.nix]: The blank buildInputs?

Comment (joachifm) [inline pkgs/development/compilers/rgbds/default.nix]: Yes

Comment (matthewbauer) [inline pkgs/development/compilers/rgbds/default.nix]: Not really, just left over from the template.

File: pkgs/development/compilers/wla-dx/default.nix (hunk 1 above)
Comment (jagajaga) [inline pkgs/development/compilers/wla-dx/default.nix]: https://nixos.org/nixpkgs/manual/#sec-package-naming
> If package is fetched from git's commit then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also add "git" to the name - e.g., "pkgname-git-2014-09-23".

**Comment by matthewbauer** [comment]
Could someone just confirm for me that it works on a Linux system? There's no reason why it shouldn't work I just don't want to run into any weird compatibility issues. I'll be home over the weekend to get to my NixOS laptop.

**Comment by zimbatm** [comment]
:+1: nox-review is all good on NixOS

**Pull request merged**